src/Tests/dotnet-workload-install.Tests/GivenNetSdkManagedWorkloadInstall.cs

```diff
@@ -0,0 +1,255 @@
+            var packInfo = new PackInfo("Xamarin.Android.Sdk", "8.4.7", WorkloadPackKind.Sdk, Path.Combine(dotnetRoot, "packs", "Xamarin.Android.Sdk", "8.4.7"));
+            var version = "6.0.100";
+            installer.InstallWorkloadPack(packInfo, new SdkFeatureBand(version));
+
+            var mockNugetInstaller = nugetInstaller as MockNuGetPackageInstaller;
+            mockNugetInstaller.InstallCallParams.Count.Should().Be(1);
+            mockNugetInstaller.InstallCallParams[0].ShouldBeEquivalentTo((new PackageId(packInfo.Id), new NuGetVersion(packInfo.Version)));
+            mockNugetInstaller.ExtractCallParams.Count.Should().Be(1);
+            mockNugetInstaller.ExtractCallParams[0].ShouldBeEquivalentTo((mockNugetInstaller.InstallCallResult[0], Path.Combine(dotnetRoot, "metadata", "temp", packInfo.Id + "-extracted")));
+
+            var installationRecordPath = Path.Combine(dotnetRoot, "metadata", "workloads", "InstalledPacks", packInfo.Id, packInfo.Version, version);
+            File.Exists(installationRecordPath).Should().BeTrue();
+
+            Directory.Exists(packInfo.Path).Should().BeTrue();
+        }
+
+        [Fact]
+        public void GivenManagedInstallItCanInstallSingleFilePacks()
+        {
+            var (dotnetRoot, installer, nugetInstaller) = GetTestInstaller();
+            var packInfo = new PackInfo("Xamarin.Android.Sdk", "8.4.7", WorkloadPackKind.Template, Path.Combine(dotnetRoot, "template-packs", "Xamarin.Android.Sdk.8.4.7.nupkg"));
+            var version = "6.0.100";
+            installer.InstallWorkloadPack(packInfo, new SdkFeatureBand(version));
+
+            (nugetInstaller as MockNuGetPackageInstaller).InstallCallParams.Count.Should().Be(1);
+            (nugetInstaller as MockNuGetPackageInstaller).InstallCallParams[0].ShouldBeEquivalentTo((new PackageId(packInfo.Id), new NuGetVersion(packInfo.Version)));
+            (nugetInstaller as MockNuGetPackageInstaller).ExtractCallParams.Count.Should().Be(0);
+
+            var installationRecordPath = Path.Combine(dotnetRoot, "metadata", "workloads", "InstalledPacks", packInfo.Id, packInfo.Version, version);
+            File.Exists(installationRecordPath).Should().BeTrue();
+
+            File.Exists(packInfo.Path).Should().BeTrue();
+        }
+
+        [Fact]
+        public void GivenManagedInstallItDetectInstalledPacks()
+        {
+            var (dotnetRoot, installer, _) = GetTestInstaller();
+            var packInfo = new PackInfo("Xamarin.Android.Sdk", "8.4.7", WorkloadPackKind.Sdk, Path.Combine(dotnetRoot, "packs", "Xamarin.Android.Sdk", "8.4.7"));
+            var version = "6.0.100";
+
+            // Mock installing the pack
+            Directory.CreateDirectory(packInfo.Path);
+
+            installer.InstallWorkloadPack(packInfo, new SdkFeatureBand(version));
+
+            _reporter.Lines.Should().Contain(string.Format(LocalizableStrings.WorkloadPackAlreadyInstalledMessage, packInfo.Id, packInfo.Version));
```

Comment:
Instead of (or in addition to) verifying the message, could this test check that the NuGet package installer was never called to download the package?